### ISSUE TITLE
"unused" false positive on declared variable with type annotation and no assignment

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3322]
### `main.py`

```py
def fn(value: bool):
    a: int  # hint: `a` is unused
    if value:
        a = 1
    else:
        a = 2

    return a
```

this is a common pattern if you want to explicitly declare the type before assignment if it's assigned in multiple possible spots. this declaration is not an assignment so it should not be considered unused

**Comment by ibraheemdev** [comment]
Hi, can you share what version of ty you are using? I don't see any diagnostics on the linked playground.

Edit: Sorry, I realized that it's a hover message not a file diagnostic, and the declaration is greyed out.
